CDCgov/prime-reportstream
# 6310 Support CLIA IDs in full-ELR pipeline

This PR adds support for CLIA IDs in the full-ELR pipeline.

Test Steps:
1. Build this branch
2. Run the following commands
```
./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_2_20220803-0001.hl7 --output-file build/hca-fhir.json
./prime fhirdata --input-file build/hca-fhir.json -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml
```
4. Verify the resulting HL7 has CLIA IDs (see MSH 3-6, and/or ORC 2-4, and/or OBR 2-3)

**Sample output:**
```
MSH|^~\&|MEDITECH|North Suburban Medical Center^06D0514637^CLIA|DCEED.ELR^2.16.840.1.114222.4.1.144.2.5^ISO|CDPHE^2.16.840.1.114222.4.1.144^ISO|20220803060705-0400||ORU^R01^ORU_R01|MT_COCNB_ORU_NBPHELR.1.5348467|P|2.5.1
SFT|MEDITECH, Inc.^L^^^^MEDITECH&1.3.6.1.4.1.24310&ISO^XX^^^MEDITECH|5.67|Laboratory Applicati|20220803||20220803
PID|1||A0995844505^^^^MR^COCNB~A09958445050^^^^SS^COCNB~A09958445051^^^^PI^COCNB~A09958445052^^^^AN^COCNB||Willms^Bernese^L||19660421|F||W^White^HL70005|0 Witting Circles^Suite 579^Dove Creek^CO^81324||^PRN^PH|^WPN^PH||S|1041^^HL70006|||||1^^HL70189||||||||N|||20220708110200-0600|COCNB|337915000^Homo sapiens (organism)^SCT^Human^Human^L
NK1|1|Connelly^Kin|^^^SO^SON^L|4 Precious Pines^Apt. 613^Warren^AR^71671|^^PH
PV1|1|E|F.ER|EM|||MCLSH^McLaughlin^Shaunderrios^T^^^MD|BECLE^Becker^Lesmer^F^^^MD|BECLE^Becker^Lesmer^F^^^MD|||||PR||||||03|||||||||||||||||||COCNB||DE
ORC|RE|05187084^M7586249.2^06D0514637^CLIA|22:NB:B0016908S.2^M7586249.2^06D0514637^CLIA|22:NB:B0016908S^M7586249^06D0514637^CLIA||N|||20220803040700+0000|||LEUJE^Leuschke^Jediah^^^^MD|||20220621133600+0000||N. SUBURBAN MED. CEN||||******************************^L^^^^N. SUBURBAN MED. CEN^XX^^^N. SUBURBAN MED. CEN|9191 GRANT STREET^LABORATORY^THORNTON^CO^80229^***^B^^PA|^^PH^^1^999^4504476|5600 S Quebec St Ste 312A^^Greenwood Village^CO^80111^***^B^^PA
OBR|1|05187084^M7586249.2^06D0514637^CLIA|22:NB:B0016908S.2^M7586249.2^06D0514637^CLIA|94500-6^SARS-CoV-2 RNA Resp Ql NAA+probe^LN^COVID19IHCE^Novel Coronavirus 2019 PCR^L|||20220621133600-0600|20220621133600-0600|||||HEADACHE|||LEUJE^Leuschke^Jediah^^^^MD||22:NB:B0016908Sm7586249|05187084||******************************|20220803040600-0600|||F
ORC|RE|05187084^M7586249.3|22:NB:B0016908S.3^M7586249.3|22:NB:B0016908S^M7586249||N|||20220803040700+0000|||LEUJE^Leuschke^Jediah^^^^MD|||20220621133600+0000||N. SUBURBAN MED. CEN||||******************************^L^^^^N. SUBURBAN MED. CEN^XX^^^N. SUBURBAN MED. CEN|9191 GRANT STREET^LABORATORY^THORNTON^CO^80229^***^B^^PA|^^PH^^1^999^4504476|5600 S Quebec St Ste 312A^^Greenwood Village^CO^80111^***^B^^PA
OBR|2|05187084^M7586249.3|22:NB:B0016908S.3^M7586249.3|34487-9^Influenza virus A RNA [Presence] in Unspecified specimen by NAA with probe detection^LN^INFLAPCR^INFLUENZA A VIRUS PCR^L|||20220621133600-0600|20220621133600-0600|||||HEADACHE|||LEUJE^Leuschke^Jediah^^^^MD||22:NB:B0016908Sm7586249|05187084||******************************|20220803040600-0600|||F
ORC|RE|05187084^M7586249.4|22:NB:B0016908S.4^M7586249.4|22:NB:B0016908S^M7586249||N|||20220803040700+0000|||LEUJE^Leuschke^Jediah^^^^MD|||20220621133600+0000||N. SUBURBAN MED. CEN||||******************************^L^^^^N. SUBURBAN MED. CEN^XX^^^N. SUBURBAN MED. CEN|9191 GRANT STREET^LABORATORY^THORNTON^CO^80229^***^B^^PA|^^PH^^1^999^4504476|5600 S Quebec St Ste 312A^^Greenwood Village^CO^80111^***^B^^PA
OBR|3|05187084^M7586249.4|22:NB:B0016908S.4^M7586249.4|40982-1^Influenza virus B RNA [Presence] in Unspecified specimen by NAA with probe detection^LN^INFLBPCR^INFLUENZA B VIRUS PCR^L|||20220621133600-0600|20220621133600-0600|||||HEADACHE|||LEUJE^Leuschke^Jediah^^^^MD||22:NB:B0016908Sm7586249|05187084||******************************|20220803040600-0600|||F

```
## Changes
- Added new deidentified sample from HCA's hca.default feed.  Had to fix a few fields that had invalid data (they replaced data with asterisks).
- Fixed an issue with the phone number functions throwing an exception is the phone number could not be read.
- Updated configurations to support CLIA IDs.
- Added ID related FHIR path functions

## Checklist

### Testing

### Process

## Linked Issues
- #6310

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
